assets/js/base/hooks/payment-methods/use-payment-method-interface.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+import {
+	useCheckoutContext,
+	usePaymentMethodDataContext,
+	useShippingMethodDataContext,
+} from '@woocommerce/base-context';
+import { __ } from '@wordpress/i18n';
+import { getCurrencyFromPriceResponse } from '@woocommerce/base-utils';
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	useStoreOrder,
+	useStoreCartCoupons,
+	useStoreCart,
+	useBillingData,
+} from '..';
+
+/**
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').CartTotalItem} CartTotalItem
+ */
+
+// @todo similar logic is done in `blocks/cart-checkout/cart/full-cart/index.js`
+// we should extract this to use in both places so it's consistent.
+// @todo this will need to account for `DISPLAY_PRICES_INCLUDING_TAXES`.
+/**
```

Comment:
Agree we need an abstraction for cart totals/arithmetic! 

I also think it's a good idea to add an explicit method for converting a money value to float, i.e. `parseInt( money_value, 10)` is used a lot. Though if we have a cart calculations module then perhaps that's not a big deal.

These `@todo` comments should have matching issues 📥

Reply:
> These @todo comments should have matching issues 📥

See [this comment](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1929#issuecomment-597548637]